### ISSUE TITLE
Docker dev with php 8.3 - Errors with cli

### ISSUE THREAD

**Issue by bernier-db** [ymirapp/cli#39]
Hi!
First of all, I think this project is quite awesome. I already use bref for some laravel projects, and I am quite excited to have the same possibility with wp.

Sadly, I get some errors with the cli. Was able to init the project, but cannot setup either dns zone or ssl. I get this error, though domain is indeed specified (`ymir certificate:request mydomain.com`)
                                        
```
  The Ymir API responded with errors:   
                                        
      * The domains field is required.  
```

I also have a warning saying a new version is available, but I just installed globally using composer (in the container).
`ymir --version` returns Ymir 1.6.2

Here is my dockerfile for development:

```
# Use an official PHP runtime as a parent image
FROM php:8.3-apache

# Set the working directory
WORKDIR /var/www/html

ADD --chmod=0755 https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/


# Install system dependencies
RUN apt-get clean && apt-get update && apt-get install -y \
    git \
    zip \
    unzip \
    curl \
    default-mysql-client

RUN install-php-extensions gd xdebug zip mysqli

# Install Composer
RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer

# Install AWS CLI
RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
    unzip awscliv2.zip && \
    ./aws/install

# Install Ymir CLI
RUN composer global require ymirapp/cli
RUN export PATH="$PATH:$HOME/.composer/vendor/bin"

# Enable Apache mod_rewrite
RUN a2enmod rewrite

#install wordpress cli
RUN curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar && \
    chmod +x wp-cli.phar && \
    mv wp-cli.phar /usr/local/bin/wp

# create dev user
RUN useradd -m -s /bin/bash dev

# Copy the current directory contents into the container at /var/www/html
COPY . /var/www/html

# Make port 80 available to the world outside this container
EXPOSE 80

# Start Apache in the foreground
CMD ["apache2-foreground"]

```

Here is the docker-compose file :

```
version: '3.8'

services:
  web:
    build: ./docker
    volumes:
      - .:/var/www/html
    ports:
      - "8080:80"
    depends_on:
      - db
    environment:
      - DB_HOST=db
      - DB_USER=root
      - DB_PASSWORD=secret
      - DB_NAME=wordpress
    command: /bin/bash -c "apache2-foreground"
  db:
    image: mysql:8.3
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: secret
      MYSQL_DATABASE: wordpress
    volumes:
      - db_data:/var/lib/mysql

volumes:
  db_data:
```

Any insight would be appreciated, thanks!

**Comment by bernier-db** [comment]
Alright, turns out latest cli version is 1.45.

Had to install it specifying the version number : `composer global require ymirapp/cli:v1.45.0`

Also was missing a few packages, so changing the docker file deps install to this : 
`RUN install-php-extensions gd xdebug zip mysqli sodium ftp`

**Comment by carlalexander** [comment]
Thanks @bernier-db! Actually, I had a guide on how to install gd. You needed:

```Dockerfile
FROM --platform=linux/x86_64 ymirapp/php-runtime:php-<PHP_VERSION>

ENTRYPOINT []

CMD ["/bin/sh", "-c", "/opt/bootstrap"]

RUN LD_LIBRARY_PATH= yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
RUN LD_LIBRARY_PATH= yum install -y https://rpms.remirepo.net/enterprise/remi-release-7.rpm
RUN LD_LIBRARY_PATH= yum install -y yum-utils
RUN LD_LIBRARY_PATH= yum-config-manager --enable remi-php<PHP_VERSION>
RUN LD_LIBRARY_PATH= yum install -y php<PHP_VERSION>-php-gd

RUN echo "extension=/opt/remi/php<PHP_VERSION>/root/usr/lib64/php/modules/gd.so" >> /opt/ymir/etc/php/conf.d/php.ini

COPY runtime/src /opt/src

COPY . /var/task
```

But if this works, that's easier! Did you do it on arm?  Also some of the extensions should be there like sodium.

**Comment by bernier-db** [comment]
Thanks for the reply @carlalexander 

I must have missed your guide. I skipped over the docker documentation since I'm not planning to deploy with my own container at this point.

I'ill give it a try. As I only started the project, I only installed WP and deployed to make sure the basic works. I might get surprises coming up, but it's good to know you have this guide.


> Did you do it on arm?

I'm developing on an Intel Mac. So no, not ARM.
